DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,128 @@
+# Developer guide
+
+This document describes how to set up a development environment for Calico, as well as how to build and test development code.
+
+This guide is broken into the following main sections:
+
+- [Building the code](#building-the-code)
```

Comment:
Might be a little more clear if you refer to it as "building / deploying images" instead of "code"?

idk maybe that's just my opinion